### PR TITLE
Handle persisted checkpoints with missing line attributions

### PR DESCRIPTION
**Pull request by zhangzimingmmz**
## Summary
- fall back from missing persisted `line_attributions` to character attribution conversion
- skip stale entries when no line attribution can be recovered instead of aborting the whole authorship note generation
- add a regression test for a stale zero-width AI attribution entry in a persisted working log

## Root cause
`VirtualAttributions::from_persisted_working_log` treated checkpoint entries with character attributions but empty `line_attributions` as fatal when the checkpoint had AI additions. A stale/no-op entry could therefore fail post-commit authorship generation for the entire commit, even when other files in the commit had valid AI attribution data. Other reconstruction paths already convert character attributions back to line attributions and skip empty results; this makes the persisted path behave consistently.

## Verification
- `cargo test --test integration test_stale_zero_width_checkpoint_entry_does_not_abort_persisted_working_log -- --nocapture`
- `cargo fmt`
- `cargo clippy --all-targets -- -D warnings`
- `git diff --check`

Note: I ran cargo/rustc through the installed stable toolchain path locally because the rustup shim on this machine hangs without an explicit default toolchain in the test harness environment.

---

<a href="https://app.devin.ai/review/git-ai-project/git-ai/pull/1593" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Comment by zhangzimingmmz** [comment]
Adding a little more context for maintainers:

This came from a real attribution collection failure where a commit had valid AI-generated changes, but post-commit authorship note generation aborted with:

```text
checkpoint entry for <file> missing persisted line attributions
```

The problematic checkpoint entry was stale/no-op-like: it still had character-level attribution data, but `line_attributions` was empty. Because `from_persisted_working_log` treated that as fatal, one bad entry prevented the entire commit from getting an authorship note, causing the commit to show up as unknown/unattributed even though other files had valid AI attribution data.

This PR makes the persisted reconstruction path match the more tolerant behavior used by the other reconstruction paths: try converting character attributions back to line attributions, and if no lines can be recovered, skip the stale entry instead of aborting the whole note generation.

File: tests/integration/simple_additions.rs [hunk 1]
```diff
@@ -1786,6 +1788,69 @@ fn test_ai_generated_file_then_human_full_rewrite() {
...
+/// line attribution must not abort note generation for the whole commit.
+#[test]
+fn test_stale_zero_width_checkpoint_entry_does_not_abort_persisted_working_log() {
+    let repo = TestRepo::new();
+    let feature_path = repo.path().join("feature.rs");
+
+    fs::write(&feature_path, "fn main() {}\n").unwrap();
+    repo.git_ai(&["checkpoint", "mock_known_human", "feature.rs"])
+        .unwrap();
+    repo.stage_all_and_commit("base").unwrap();
```
[the remark this answers is not in the document]
Comment (zhangzimingmmz) [reply tests/integration/simple_additions.rs:1801]: Fixed by adding a post-base-commit line-level assertion and reusing the same TestFile for the final assertion.

**Comment by svarlamov** [comment]
Thank you for the contribution @zhangzimingmmz -- very real bug and solid solution. Excited to see more contributions from you in the future and please let us know if there's anything else you find!
